hazelcast/src/main/java/com/hazelcast/internal/metrics/managementcenter/MetricsResultSet.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.metrics.managementcenter;
+
+import javax.annotation.Nonnull;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.hazelcast.internal.metrics.managementcenter.MetricsCompressor.decompressingIterator;
+
+public class MetricsResultSet {
+
+    private final long nextSequence;
+    private final List<MetricsCollection> collections;
+
+    public MetricsResultSet(ConcurrentArrayRingbuffer.RingbufferSlice<Map.Entry<Long, byte[]>> slice) {
+        this.nextSequence = slice.nextSequence();
+        this.collections = slice.stream()
+                                .map(e -> new MetricsCollection(e.getKey(), e.getValue()))
+                                .collect(Collectors.toList());
+    }
+
+    /**
+     * The next sequence to read from.
+     */
+    public long nextSequence() {
+        return nextSequence;
+    }
+
+    public List<MetricsCollection> collections() {
+        return collections;
+    }
+
+    /**
+     * Deserializing iterator for reading metrics
+     */
+    public static final class MetricsCollection implements Iterable<Metric> {
+
+        private final long timestamp;
```

Comment:
Shouldn't this be `sequence` instead of `timestamp` for the sake of consistency?

Reply:
The ringbuffer is typed with `Map.Entry<Long, byte[]>` where the key is a timestamp taken in `ManagementCenterPublisher#whenComplete`. That's where the timestamp is from.

Reply:
Yes, I know that. It's just strange that the same thing (timestamp) is called `sequence` in other APIs, like `HazelcastClientInstanceImpl#readMetricsAsync`'s arguments or `MetricsResultSet#nextSequence`.

Reply:
They're not the same thing, sequence is the sequence of the metrics in the ringbuffer, timestamp is the system time when it's collected.

Reply:
> They're not the same thing, sequence is the sequence of the metrics in the ringbuffer, timestamp is the system time when it's collected.

They're tightly coupled at the moment. As an example, `startSequence` argument in `HazelcastClientInstanceImpl#readMetricsAsync` really means `startTimestamp`/`fromTimestamp` in terms of internal processing and it affects what is returned in `MetricsCollection`s, including `timestamp` field values.

I'm totally OK if we leave as is, but I'd prefer to use term `timestamp` in all "external" APIs to avoid confusion.

Reply:
A follow-up. I've double checked the `ConcurrentArrayRingbuffer` implementation and now I see that `sequence` is position in the ring buffer, as you said. Thanks for the clarification @cangencer!